### PR TITLE
bump locked ansible-lint version to 4.3.5; suppress role name lint warning

### PR DESCRIPTION
**Pull request by richm**
unfreeze the ansible-lint version to use the latest version 4.3.5 which
is compatible with the latest ansible
This means we need to suppress the error about the role name, and
the error about missing `mode` in `file` and `template` usage.

### CODE REVIEW

**Review by nhosoi — APPROVED** [review]
lgtm

File: tox.ini [hunk 1]
```diff
@@ -108,7 +108,7 @@ commands =
 [molecule_common]
 changedir = {toxinidir}
 deps =
-    ansible-lint==4.2.0
```
Comment (pcahyna) [inline tox.ini:111]: Should we be more conservative and keep ansible-lint locked at a newer, known good version? (that would be 4.3.5 I suppose).

**Comment by richm** [comment]
please review

**Comment by richm** [comment]
> I'm curious. The failure of Travis CI randomly occurs with python 2.6? It seems it passed in the previous patch although it does not look the version of ansible-lint got involved...

we should probably drop python 2.6 from .travis.yml

**Comment by pcahyna** [comment]
> The failure of Travis CI randomly occurs with python 2.6? It s

The error is:
```
Collecting tox

/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#snimissingwarning.

  SNIMissingWarning

/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.

  InsecurePlatformWarning

  Could not fetch URL https://pypi.python.org/simple/tox/: There was a problem confirming the ssl certificate: [Errno 1] _ssl.c:499: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed - skipping
```
As it is a problem with SSL, it probably depends on what mirror gets chosen and what requirement on SSL support it has, which is essentially random.

**Pull request merged**